docs/pages/kubernetes-access/guides/dynamic-registration.mdx

```diff
@@ -0,0 +1,458 @@
+spec:
+  allow:
+    rules:
+    - resources:
+      - kube_cluster
+      verbs:
+      - list
+      - create
+      - read
+      - update
+      - delete
+version: v5
+```
+
+Retrieve your user:
+
+```code
+$ TELEPORT_USER=myuser
+$ tctl get user/${TELEPORT_USER?} > user.yaml
+```
+
+Add `kube-manager` to your Teleport user's list of roles:
+
+```diff
+   roles:
+   - access
+   - auditor
++  - kube-manager
+```
+
+Apply your changes:
+
+```code
+$ tctl create -f kube-manager.yaml
+$ tctl create -f user.yaml
+```
+
+Log out of your cluster and log in again to assume the new role.
+
+## Step 3/3. Manage dynamic Kubernetes cluster resources
+
+Now that your Teleport user has permissions to manage Kubernetes cluster
+resources, we will show you how to create, list, update, and delete them. 
+
+### Create a kubeconfig
+
+In this section, you will create a Kubernetes `Config` resource, or kubeconfig,
+that your Teleport cluster will use to authenticate to your Kubernetes cluster.
+
+When you signed into Teleport earlier in this guide, `tsh` may have changed your
+Kubernetes context to one based on your Teleport cluster, so make sure you
+update your Kubernetes context to match the cluster you would like to connect to
+Teleport: 
+
+```code
+$ kubectl config get-contexts
+# Assign CONTEXT_NAME to your chosen context
+$ kubectl config use-context CONTEXT_NAME
+```
+
+On your workstation, download Teleport's `get-kubeconfig.sh` script, which you
+will use to generate the kubeconfig:
+
+```code
+$ curl -OL \
+https://raw.githubusercontent.com/gravitational/teleport/master/examples/k8s-auth/get-kubeconfig.sh
+```
+
+The script creates a service account for the Teleport Kubernetes Service that
+can get Kubernetes pods as well as impersonate users, groups, and other service
+accounts. The Teleport Kubernetes Service uses this service account to manage
+access to resources in your Kubernetes cluster. The script also ensures that
+there is a Kubernetes `Secret` in your cluster to store service account
+credentials. 
+
+`get-kubeconfig.sh` creates a namespace called `teleport` for the resources it
+deploys, though you can choose a different name by assigning the
+`TELEPORT_NAMESPACE` environment variable in the shell where you run the script. 
+
+After creating resources, `get-kubeconfig.sh` writes a new kubeconfig to a file
+called `kubeconfig` in the directory where you run the script.
+
+Execute the following command:
+
+```code
+$ bash get-kubeconfig.sh
+```
+
+The script is successful if you see this message:
+
+```text
+Done!
+```
+
+Ignore the script's instructions to copy the generated kubeconfig file to the
+Teleport Proxy Service. In the next section, we will show you how to use the
+kubeconfig file when creating a dynamic `kube_cluster` resource.
+
+### Create a Kubernetes cluster resource
+
+Define a `kube_cluster` resource with the following content in a file called
+`kube_cluster.yaml`:
+
+```yaml
+kind: kube_cluster
+version: v3
+metadata:
```

Comment:
Maybe it makes sense to mention that we can add labels and dynamic labels to the configuration here. Useful for Teleport RBAC